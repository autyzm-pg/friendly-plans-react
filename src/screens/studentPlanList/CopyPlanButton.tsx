import { FlatButton } from 'components';
import { i18n } from 'locale';
import React from 'react';
import { StyleSheet } from 'react-native';
import { dimensions, palette } from 'styles';

export class CopyPlanButton extends React.PureComponent {
  render() {
    return (
      <FlatButton
        title={i18n.t('planList:copyPlan')}
        icon={{
          name: 'ios-copy',
          type: 'ionicon',
          color: palette.primary,
          size: 14,
        }}
        buttonStyle={styles.button}
        titleStyle={styles.title}
      />
    );
  }
}

const styles = StyleSheet.create({
  button: {
    margin: 15,
  },
  title: {
    marginLeft: 3,
    textTransform: 'uppercase',
  },
});
