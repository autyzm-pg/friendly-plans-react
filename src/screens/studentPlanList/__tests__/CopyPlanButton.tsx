import { shallow } from 'enzyme';
import React from 'react';
import { CopyPlanButton } from '../CopyPlanButton';

describe('CopyPlanButton', () => {
  it('renders correctly', () => {
    const copyPlanButton = shallow(<CopyPlanButton />);

    expect(copyPlanButton).toMatchSnapshot();
  });
});
