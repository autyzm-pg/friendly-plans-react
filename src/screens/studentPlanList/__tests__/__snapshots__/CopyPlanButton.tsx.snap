// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyPlanButton renders correctly 1`] = `
<FlatButton
  buttonStyle={
    Object {
      "margin": 15,
    }
  }
  icon={
    Object {
      "color": "#6654EB",
      "name": "ios-copy",
      "size": 14,
      "type": "ionicon",
    }
  }
  title="Copy existing plan"
  titleStyle={
    Object {
      "marginLeft": 3,
      "textTransform": "uppercase",
    }
  }
/>
`;
